include/clover_field_order.h

```diff
@@ -519,49 +558,60 @@ namespace quda {
        pointer arithmetic for huge allocations (e.g., packed set of
        vectors).  Default is to use 32-bit pointer arithmetic.
     */
-    template <typename Float, int length, int N, bool add_rho=false, bool huge_alloc=false>
-    struct FloatNOrder {
-      using Accessor = FloatNOrder<Float, length, N, add_rho, huge_alloc>;
-      using real = typename mapper<Float>::type;
-      typedef typename VectorType<Float, N>::type Vector;
-      typedef typename AllocType<huge_alloc>::type AllocInt;
-      typedef float norm_type;
-      static const int M = length / (N * 2); // number of short vectors per chiral block
-      static const int block = length / 2;   // chiral block size
-      Float *clover;
-      norm_type *norm;
-      const AllocInt offset; // offset can be 32-bit or 64-bit
-      const AllocInt norm_offset;
-	const int volumeCB;
+      template <typename Float, int length, int N, bool add_rho = false,
+                bool enable_reconstruct_ = clover::reconstruct(), bool huge_alloc = false>
+      struct FloatNOrder {
+        static constexpr bool enable_reconstruct = enable_reconstruct_;
+        using Accessor = FloatNOrder<Float, length, N, add_rho, enable_reconstruct, huge_alloc>;
+        using real = typename mapper<Float>::type;
+        typedef typename VectorType<Float, N>::type Vector;
+        typedef typename AllocType<huge_alloc>::type AllocInt;
+        typedef float norm_type;
+        static constexpr int Ns = 4;
+        static constexpr int Nc = 3;
```

Comment:
Can you place a static_assert here on 3?

Reply:
84359ab